static/sw.js

```diff
@@ -0,0 +1,400 @@
+const CACHE_VERSION = 2.3;
+
+const BASE_CACHE_FILES = [
+    '/js/fuse.min.94c78ad70b02749822921660cf4e9f0b3701bc0680c421afb784a78228de0275.js',
+    '/js/helper/getParents.min.1618c696be7c98933f9a92677f518b512a74e55bdbb976b09936b4182e93181b.js',
+    '/css/main.min.css',
+    '/js/helper/fadeinout.min.93a331f96194789a542f33690bbe4f0c102c7e78ffc018217f5a1c33010bad91.js',
+    '/logo.png',
+    '/js/lazysizes.min.31dd6a2d3a1ec0f78a8df007535cf23f03aeb5c70f026e6d6a19dac3b3acc340.js',
+    '/fonts/montserrat-bold.woff2',
+    '/js/enquire.min.dfb99dee1e029d51d6cfb672d847929890b1585402de17f5ed092edd72a688b4.js',
+    '/fonts/montserrat-regular.woff2',
+    '/fonts/merriweather-regular.woff2',
+    
+];
+
+const OFFLINE_CACHE_FILES = [
+    '/js/fuse.min.94c78ad70b02749822921660cf4e9f0b3701bc0680c421afb784a78228de0275.js',
+    '/js/helper/getParents.min.1618c696be7c98933f9a92677f518b512a74e55bdbb976b09936b4182e93181b.js',
+    '/css/main.min.css',
+    '/js/helper/fadeinout.min.93a331f96194789a542f33690bbe4f0c102c7e78ffc018217f5a1c33010bad91.js',
+    '/logo.png',
+    '/js/lazysizes.min.31dd6a2d3a1ec0f78a8df007535cf23f03aeb5c70f026e6d6a19dac3b3acc340.js',
+    '/fonts/montserrat-bold.woff2',
+    '/offline/'
+];
+
+const NOT_FOUND_CACHE_FILES = [
+    '/js/fuse.min.94c78ad70b02749822921660cf4e9f0b3701bc0680c421afb784a78228de0275.js',
+    '/js/helper/getParents.min.1618c696be7c98933f9a92677f518b512a74e55bdbb976b09936b4182e93181b.js',
+    '/css/main.min.css',
+    '/js/helper/fadeinout.min.93a331f96194789a542f33690bbe4f0c102c7e78ffc018217f5a1c33010bad91.js',
+    '/logo.png',
+    '/js/lazysizes.min.31dd6a2d3a1ec0f78a8df007535cf23f03aeb5c70f026e6d6a19dac3b3acc340.js',
+    '/fonts/montserrat-bold.woff2',
+    '/404.html'
+];
+
+const OFFLINE_PAGE = '/offline/';
+const NOT_FOUND_PAGE = '/404.html';
+
+const CACHE_VERSIONS = {
+    assets: 'assets-v' + CACHE_VERSION,
+    content: 'content-v' + CACHE_VERSION,
+    offline: 'offline-v' + CACHE_VERSION,
+    notFound: '404-v' + CACHE_VERSION,
+};
+
+// Define MAX_TTL's in SECONDS for specific file extensions
+const MAX_TTL = {
+    '/': 3600,
+    html: 3600,
+    json: 86400,
+    js: 86400,
+    css: 86400,
+};
+
+const CACHE_BLACKLIST = [
+    //(str) => {
+       // return !str.startsWith('https://yourwebsite.com');
+    //},
+];
```

Comment:
I commented this lines, but users can use this. This is essentially bans anything not served under this domain. This means things like external advertisements won't work in offline mode, which is perfectly fine.